src/lib/journey/journey.store.ts

```diff
@@ -4,72 +4,57 @@ import {
   FRLoginFailure,
   FRLoginSuccess,
   StepType,
-  TokenManager,
-  UserManager,
 } from '@forgerock/javascript-sdk';
-import { get, writable, type Writable } from 'svelte/store';
+import type { Step, StepOptions } from '@forgerock/javascript-sdk/lib/auth/interfaces';
+import { writable, type Writable } from 'svelte/store';
 
-import type { StringDict } from '$lib/interfaces';
 import { htmlDecode } from '$lib/journey/utilities/decode.utilities';
-import { email, isAuthenticated, fullName } from '$lib/user/user.store';
 
-export interface InitObject {
-  step: Writable<StepTypes>;
-  getStep: (prevStep?: StepTypes) => Promise<void>;
-  failureMessage: Writable<string | null>;
-  submittingForm: Writable<boolean>;
+export interface JourneyStore extends Pick<Writable<JourneyStoreValue>, 'subscribe'> {
+  next: (prevStep?: StepTypes, nextOptions?: StepOptions) => void;
+  reset: () => void;
 }
-interface Options {
-  tree?: string;
-  query?: StringDict<string>;
+export interface JourneyStoreValue {
+  completed: boolean;
+  // TODO: Think about turning this into an object with code, message and step
+  error: {
+    code: number | null;
+    message: string | null;
+    step: Step | undefined;
+  } | null;
+  loading: boolean;
+  step: StepTypes;
+  successful: boolean;
+  response: Step | null | undefined;
 }
-interface User {
-  family_name: string;
-  given_name: string;
-  email: string;
-  name: string;
-  updated_at: number;
-  sub: string;
+export interface Response {
+  tokenId: string;
 }
 export type StepTypes = FRStep | FRLoginSuccess | FRLoginFailure | null;
 
-export async function initialize(journey?: string): Promise<InitObject> {
-  const step: Writable<StepTypes> = writable(null);
-  const failureMessage: Writable<string | null> = writable(null);
-  const options: Options = {};
-  const submittingForm: Writable<boolean> = writable(false);
+export function initialize(initOptions?: StepOptions): JourneyStore {
```

Comment:
This just initializes the journey store and provides a closure for configuration values. It does not call AM, just sets everything up.